R/set_args.R

```diff
@@ -45,4 +45,112 @@ set_inherited_args <- function(fun.name, inherited.args, ..., package='gsplot'){
+      names.index <- which(names(paramsAll) %in% pt.names)
+      pt.names.index <- which(pt.names %in% names(paramsAll))
+      names(paramsAll)[names.index] <- paste0("pt.", pt.names)[pt.names.index]
+      
+      if(!is.null(type) && type %in% c('l','b','o')){
+          if(is.null(paramsAll$lwd)){paramsAll$lwd <- 1}
+          if(is.null(paramsAll$lty)){paramsAll$lty <- 1}
+          if(type=='l'){paramsAll$pch <- NA}
+      } else if(!is.null(type) && type %in% c('c','h','s','S')){
+          paramsAll$pch <- NA
+          if(is.null(paramsAll$lwd)){paramsAll$lwd <- 1}
+          if(is.null(paramsAll$lty)){paramsAll$lty <- 1}
+      } else if(!is.null(type) && type %in% c('n')){
+          paramsAll$pch <- NA
+      } 
+      
+      fun.specific <- list(fill=quote(par("bg")),
+                           border=quote(par("bg")),
+                           lty=NA,
+                           lwd=NA,
+                           pch=1,
+                           density=NA,
+                           pt.bg=quote(par("bg")),
+                           pt.cex=par("cex"),
+                           pt.lwd=par("lwd"),
+                           text.col=par("col"),
+                           text.font=1)
+
+    } else if (fun.name %in% c("lines", "abline", "arrows", "segments")) {
+      
+      if(!is.null(type) && type %in% c('p', 'b','o')){
+          if(is.null(paramsAll$pch)){paramsAll$pch <- 1}
+          if(is.null(paramsAll$pt.bg)){paramsAll$pt.bg <- quote(par("bg"))}
+          if(is.null(paramsAll$pt.cex)){paramsAll$pt.cex <- par("cex")}
+          if(is.null(paramsAll$pt.lwd)){paramsAll$pt.lwd <- par("lwd")}
+          if(type=='p'){
+            paramsAll$lty <- NA
+            paramsAll$lwd <- NA
+          }
+      } else if(!is.null(type) && type %in% c('n')){
+          paramsAll$lty <- NA
+          paramsAll$lwd <- NA
+      }
+      
+      fun.specific <- list(fill=quote(par("bg")),
+                           border=quote(par("bg")),
+                           lty=1,
+                           lwd=1,
+                           pch=NA,
+                           density=NA,
+                           pt.bg=NA,
+                           pt.cex=NA,
+                           pt.lwd=NA,
+                           text.col=par("col"),
+                           text.font=1)
+      
+    } else if (fun.name %in% c("polygon", "rect")) {
+      names.index <- which(names(paramsAll) %in% c("col"))
+      names(paramsAll)[names.index] <- "fill"
+      
+      fun.specific <- list(fill=quote(par("bg")),
+                           border=par("fg"),
+                           lty=NA,
+                           lwd=NA,
+                           pch=NA,
+                           density=NA,
+                           pt.bg=NA,
+                           pt.cex=NA,
+                           pt.lwd=NA,
+                           text.col=par("col"),
+                           text.font=1)
+    }
```

Comment:
a lot of this is the same - have one `fun.specific` that has defaults and then change them as necessary for each graphics type